rbac/management/cache.py

```diff
@@ -34,6 +34,96 @@ def connection(self):
                 raise
         return self._connection
 
+    @contextlib.contextmanager
+    def delete_handler(self, err_msg):
+        """Handle delete events."""
+        try:
+            yield
+        except exceptions.RedisError:
+            logger.exception(err_msg)
+
+    def get_from_redis(self, index):
+        """Get object from redis based on index."""
+        obj = self.connection.hget(*(self.key_for(index[0]), index[1]))
+        if obj:
+            return json.loads(obj)
+
+    def get_cached(self, index, error_message):
+        """Get cached object from redis, throw error if there is any."""
+        try:
+            return self.get_from_redis(index)
+        except exceptions.RedisError:
+            logger.exception(error_message)
+        return None
+
+    def delete_cached(self, index, obj_name):
+        """Delete cache from redis."""
+        err_msg = f"Error deleting {obj_name} for {index}"
+        with self.delete_handler(err_msg):
+            logger.info(f"Deleting {obj_name} cache for {index}")
+            self.connection.delete(self.key_for(index))
+
+    def set_cache(self, pipe, index, item):
```

Comment:
Should this be the default behavior, or should we leave the existing logic for `AccessCache` in that class/method? Just wondering if this is "general" enough to be reused, or if the classes should implement it themselves since we have different methods to set (`set` vs `hset`)?

Reply:
👍I am moving the implementation out